src/components/Summary/AddNewScoreRedux/index.js

```diff
@@ -0,0 +1,133 @@
+import React, { PureComponent } from 'react';
+import { Rate, Row, Col, Divider, Typography } from 'antd';
+
+import reduxForm from 'redux-form/es/reduxForm';
+import Field from 'redux-form/es/Field';
+import { Button, message, Form } from 'antd';
+import createComment from 'utils/comment/comment';
+const { Text, Link } = Typography;
+
+//
+const FieldizeRate = () => ({
+  input,
+  meta,
+  children,
+  hasFeedback,
+  label,
+  ...rest
+}) => {
+  const FormItem = Form.Item;
+  const hasError = meta.touched && meta.invalid;
+  return (
+    <FormItem
+      label={label}
+      validateStatus={hasError ? 'error' : 'success'}
+      hasFeedback={hasFeedback && hasError}
+      help={hasError && meta.error}
+    >
+      <Rate style={{}} allowHalf {...input} {...rest} children={children} />
+    </FormItem>
+  );
+};
+const RateScore = FieldizeRate();
+//Reference: https://codesandbox.io/s/jzyl70wpk?file=/SimpleForm.js:611-1048
+
+class AddNewScoreFormRedux extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { DisableSubmit: true };
+    this.submitForm = this.props.handleSubmit(async data => {
+      const id = this.props.questionid;
+      const params = {
+        commentRecordId: id,
+        author: data.name,
+        quality: data.quality,
+        hint: data.hint,
+        completeness: data.completeness,
+        tags: 'no more comment',
+        content: data.comment,
+      };
+      await createComment(params);
+      message.success('Add Overall Score successfully');
+      this.props.reset();
+    });
+  }
+  onDetectNullAuthor = e => {
```

Comment:
Suggest checking the redux form value of the author and three scores.

Your method is working fine but it will be executed every time when onChange is triggered.

Using `redux` `connect` to your component with `redux-form`'s `getFormValues` then you can get all the form state values.

Reply:
That's so helpful! Thank you

Reply:
I tried several methods like get.state or useStore Hook(not supported in React 16.8), this really help